plaidml/testenv.h

```diff
@@ -77,6 +79,57 @@ class TestFixture : public ::testing::Test {
     exec::Binder(program).compile()->run();
 #endif
   }
+
+  // Sets a skip if the current target is the specified target,
+  // returning true iff the current test should be skipped.
+  //
+  // N.B. The test itself continues to execute, and should use
+  //      IsSkipped() or the result of this method to exit when
+  //      necessary.  This is useful for tests whose output will be
+  //      processed by LLVM FileCheck.
+  //
+  // For example, if the test's output is being checked against its
+  // source by FileCheck, one might want to use:
+  //
+  //     setIsSkipped("some_target");
+  //
+  //     // Code that emits some program, often via makeProgram()
+  //
+  //     // CHECK-LABEL: MyTest
+  //
+  //     if (IsSkipped()) {
+  //       return;
+  //     }
+  //
+  //     // Code to run the program and check its output.
+  //
+  // On the other hand, if the test's output isn't being checked
+  // against its source by FileCheck, one can simply use:
+  //
+  //    if (setIsSkipped("some_target")) {
+  //      return;
+  //    }
+  //
+  bool setSkipOnTarget(const char* target) {
+    if (Settings::get("PLAIDML_TARGET") == target) {
+      [target]() -> void {
+        GTEST_SKIP() << "Test skipped on " << target;
+      }();
```

Comment:
This seems strange, why make a lambda and then immediately call it?

Reply:
GTEST_SKIP() includes a `return`.